plugins/scaffolder-backend/src/service/router.ts

```diff
@@ -143,81 +147,19 @@ export interface RouterOptions {
   taskBroker?: TaskBroker;
   additionalTemplateFilters?: Record<string, TemplateFilter>;
   additionalTemplateGlobals?: Record<string, TemplateGlobal>;
-  permissions?: PermissionEvaluator;
+  permissions?: PermissionsService;
   permissionRules?: Array<
     TemplatePermissionRuleInput | ActionPermissionRuleInput
   >;
-  identity?: IdentityApi;
+  auth?: AuthService;
+  httpAuth?: HttpAuthService;
+  discovery: DiscoveryService;
 }
 
 function isSupportedTemplate(entity: TemplateEntityV1beta3) {
   return entity.apiVersion === 'scaffolder.backstage.io/v1beta3';
 }
 
-/*
- * @deprecated This function remains as the DefaultIdentityClient behaves slightly differently to the pre-existing
- * scaffolder behaviour. Specifically if the token fails to parse, the DefaultIdentityClient will raise an error.
- * The scaffolder did not raise an error in this case. As such we chose to allow it to behave as it did previously
- * until someone explicitly passes an IdentityApi. When we have reasonable confidence that most backstage deployments
- * are using the IdentityApi, we can remove this function.
- */
-function buildDefaultIdentityClient(options: RouterOptions): IdentityApi {
```

Comment:
I think it's best to keep this in place, so we don't need to worry about whether removing this is breaking or not. Unless you're confident that removing this won't break any existing adopters that aren't currently passing an identity API. It would essentially be

```ts
createLegacyAuthAdapters({...options, identity: options?.identity ?? buildDefaultIdentityClient(options));
```

Reply:
Done